core/src/test/java/org/jdbi/v3/core/qualifier/QualifiedConstructorParamThing.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.qualifier;
+
+import java.util.Objects;
+
+public class QualifiedConstructorParamThing {
+    private final int id;
+    private final String name;
+
+    public QualifiedConstructorParamThing(int id, @Reversed String name) {
+        this.id = id;
+        this.name = name;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        QualifiedConstructorParamThing that = (QualifiedConstructorParamThing) o;
+        return id == that.id
+            && Objects.equals(name, that.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, name);
+    }
+
+    @Override
+    public String toString() {
+        return "QualifiedConstructorParamThing{"
```

Comment:
ugh, manual json-like tostring building... Can we include a tool like ToStringBuilder.reflectionToString to squash all of this debug tostring building logic?

Reply:
should probably go in a separate pr

Reply:
This is all IDE-generated code.

I've actually thought about using lombok to simplify our tests. Wouldn't hurt to have some tests to capture lombok common usages too.